stubs/click/click/decorators.pyi

```diff
@@ -71,7 +71,7 @@ def argument(
     expose_value: bool = ...,
     is_eager: bool = ...,
     envvar: Optional[Union[str, List[str]]] = ...,
-    autocompletion: Optional[Callable[[Any, List[str], str], List[Union[str, Tuple[str, str]]]]] = ...,
+    autocompletion: Optional[Callable[[Context, List[str], str], Iterable[Union[str, Tuple[str, str]]]]] = ...,
```

Comment:
Click itself [annotates this as List](https://github.com/pallets/click/blob/fc4ebb86c8e5fe8e7bdb3d1516f286f16865f538/src/click/core.py#L2030). Maybe we should do that too.

Those type annotations were added about a week ago. I will ask click devs whether they are going to type-annotate everything, which might mean that it's not a good idea to improve the stubs in typeshed.

Also, [autocompletion argument seems to be deprecated](https://github.com/pallets/click/blob/fc4ebb86c8e5fe8e7bdb3d1516f286f16865f538/src/click/core.py#L2058-L2067)

Reply:
Yes, it's deprecated starting 8.x which isn't released. But the replacement doesn't exist yet in 7.1.2 (last release).

I think `List` is too strong a requirement to impose on user code. But maybe that discussion should be had in `click` itself?